tox.ini

```diff
@@ -17,3 +18,10 @@ commands =
       -W ignore:'pkg_resources is deprecated as an API':DeprecationWarning \
       -m coverage run \
       -m pytest -p no:randomly {posargs:tests}
+
+[testenv:lint]
+description = run linters
+deps =
+    pre-commit
```

Comment:
We could pin the version here or use the `py311.txt` requirements file. If pinning here then we can drop `pre-commit` as a dependency above.